pkg/build/controller/controller.go

```diff
@@ -79,21 +81,74 @@ func (bc *BuildController) CancelBuild(build *buildapi.Build) error {
 // HandleBuild deletes pods for cancelled builds and takes new builds and puts
 // them in the pending state after creating a corresponding pod
 func (bc *BuildController) HandleBuild(build *buildapi.Build) error {
-	glog.V(4).Infof("Handling build %s/%s", build.Namespace, build.Name)
+	glog.V(4).Infof("Handling build %s/%s (%s)", build.Namespace, build.Name, build.Status.Phase)
+
+	// If we receive build that reached its final state and the build was created
+	// from a serial build configuration, tag the next build in the queue for
+	// execution. That will produce event and the build should get executed
+	// immediately.
+	var nextBuild *buildapi.Build
+	schedulePolicy := buildutil.BuildSchedulePolicy(build)
+	if schedulePolicy == buildapi.BuildSchedulePolicySerial {
```

Comment:
retry controller runs for every build, including the serial builds... that means if the queue is free (no builds are running) retry controller will call HandleBuild on the each serial build... if it hits the first serial build, it will be executed :-) It will just not be that fast I guess because you will have to wait till retry controller kicks in (which might take couple seconds)...

I think that is fine for now?